File: frontend/lib/models/Variante.dart
class Variante {
  final int varianteId;
  final String via;
  final int? grau;
  final String? crux;
  final String? aid;
  final String? duracao;
  final String? exposicao;
  final int? extensao;
  final String? conquistadores;
  final String? data;
  final int? sourceId;

  Variante({
    required this.varianteId,
    required this.via,
    this.grau,
    this.crux,
    this.aid,
    this.duracao,
    this.exposicao,
    this.extensao,
    this.conquistadores,
    this.data,
    this.sourceId,
  });
}

File: backend/src/entities/Variante.ts
import {
    Entity,
    PrimaryGeneratedColumn,
    Column,
    ManyToOne,
    JoinColumn
} from "typeorm";
import { Fonte } from "./Fonte";  // Assumindo que você já criou a entidade Fonte

@Entity("variantes")
export class Variante {

    @PrimaryGeneratedColumn({ name: "variante_id" })
    varianteId!: number;

    @Column({ type: "varchar", length: 255 })
    vias!: string;

    @Column({ type: "float" })
    grau!: number;

    @Column({ type: "varchar", length: 50 })
    crux!: string;

    @Column({ type: "varchar", length: 50 })
    aid!: string;

    @Column({ type: "varchar", length: 50 })
    duracao!: string;

    @Column({ type: "varchar", length: 50 })
    exposicao!: string;

    @Column({ type: "int" })
    extensao!: number;

    @Column({ type: "text" })
    conquistadores!: string;

    @Column({ type: "date" })
    data!: Date;

    @ManyToOne(() => Fonte)
    @JoinColumn({ name: "source_id" })
    fonte!: Fonte;
}
